EIPS/eip-758.md

```diff
@@ -9,30 +9,32 @@ created: 2017-11-09
+External callers presently have no way of accessing return data from Ethereum, if the function was executed via `eth_sendTransaction` or `eth_sendRawTransaction` RPC request.  Access to function return data is in many cases a desirable feature.  Making return data available to external callers also addresses the inconsistency between internal callers, which have access to return data within the context of the transaction, and external callers, which do not.  Presently, a common workaround is to log the return data, which is bad for several reasons: it contributes to chain bloat, imposes additional gas costs on the caller, and can result in unused logs being written if the externally called function involves other (internal) function calls that log their return data.  While implementing the original version of this EIP, it was decided to expand this functionality slightly to allow for external callers to be notified of their completed transactions even in the case where there is *no* return data.  This could be either because the method called doesn't return a value, or because the transaction is a simple transfer of value.
 
 ## Specification
 
 ### Subscription
-A caller who wants to be notified of return data for their transactions sends an `eth_subscribe` RPC request with the parameter `"returnData"`:
+A caller who wants to be notified when transactions of theirs complete sends an `eth_subscribe` RPC request with the first parameter `"completedTransaction"`:
 
 ```json
-{"jsonrpc": "2.0", "id": 1, "method": "eth_subscribe", "params": ["returnData"]}
+{"jsonrpc": "2.0", "id": 1, "method": "eth_subscribe", "params": ["completedTransaction", filter]}
 ```
 
+The `filter` parameter is a dictionary containing 3 optional named arguments:  `from`, `to`, and `hasReturnData`.  `from` and `to` can each either be single addresses, or a list of addresses.  They are used to filter out any transactions not sent from an address in the `from` list and sent to an address in the to list.  `hasReturnData` is a boolean--if it is specified and `true`, then notifications will be received only for completed transactions containing returnData.  If none of these optional arguments are present, then the "params" field will be `["completedTransaction", {}]` and no filter restrictions are placed on the transactions.
```

Comment:
Suggest including a specific example of a `filter`, in addition to describing it.